### DISCUSSION TITLE
Is there a way to enter a specific vae into a preset?

### DISCUSSION THREAD

**Discussion by ProtoBelisarius** [lllyasviel/Fooocus#2602] [Q&A] [answered]
The baked in vae doesnt work very well and I need to use a specific vae. 
Something like "default_vae": in the preset json doesnt work, so Im wondering if Im using the wrong name for it or if this is for some reason not possible at all.

Edit: Maybe adding these lines to the preset json might help, Im not sure though.
"default_vae": "XL.vae.safetensors",
"default_vae_approx": "XL.vae.safetensors",

**Comment by mashb1t** [comment, marked as answer]
No, there currently isn't. You can replace the default vae file with yours using the same name. Feel free to create a PR if you're considering to add vae switching as a feature.

**Reply by ProtoBelisarius** [reply]
Thank you, but how would I go about his? The folder in models/vae/ is empty(except for the vae I placed there). Is the default vae file "xlvaeapp.pth" in vae _approx/? If so, how do I extract that from a safetensors file. If not, what is the name of the default vae and what format does it have?

**Reply by ProtoBelisarius** [reply]
?

**Reply by mashb1t** [reply]
it's xlvaeapp.pth you have to swap

**Reply by ProtoBelisarius** [reply]
how do I convert a safetensors vae back to .pth?

**Reply by mashb1t** [reply]
you maybe able to just use it by renaming as both safetensors and pth are supported.

**Reply by ProtoBelisarius** [reply]
Naming it xlvaeapp.safetensors or xlvaeapp.pth.safetensors doesn't work.
How is the vae solution used by Foocus even made? The file is just 200kb in size while most vae's are around 300mb? Is there any documention about this somewhere?

**Reply by mashb1t** [reply]
There is no documentation, just rename your <vaename>.safetensors to xlvaeapp.pth

**Reply by ProtoBelisarius** [reply]
Tried that:
...
ValueError: invalid load key, '^'.
File corrupted: /home/username/Git/Fooocus/models/vae_approx/xlvaeapp.pth 
Fooocus has tried to move the corrupted file to /home/username/Git/Fooocus/models/vae_approx/xlvaeapp.pth.corrupted 
You may try again now and Fooocus will download models again.

**Reply by ProtoBelisarius** [reply]
Ok, Im not sure what helped, the vae is now in checkpoints, vae and vae_approx with its original name and it works, when I load it with the preset that defines the vae with 
"default_vae": "XL.vae.safetensors",
"default_vae_approx": "XL.vae.safetensors", 

The reason I think/thought this method didnt/doesnt work is that these were the kind of results I got:
![fail](https://github.com/lllyasviel/Fooocus/assets/103048431/ece04891-04ca-43fd-92b6-b784a791ed1c)

But maybe I was wrong? It works for me now, so Im happy. 
Have a nice day and thank you for the information.
